tasks/pipeline.py

```diff
@@ -463,6 +465,94 @@ def trigger_child_pipeline(_, git_ref, project_name, variables="", follow=True):
         print("Child pipeline finished successfully")
 
 
+def parse(commit_str):
+    lines = commit_str.split("\n")
+    title = lines[0]
+    url = "NO_URL"
+    pr_id_match = re.search(r".*\(#(\d+)\)", title)
+    if pr_id_match is not None:
+        url = f"https://github.com/DataDog/datadog-agent/pull/{pr_id_match.group(1)}"
+    author = lines[1]
+    author_email = lines[2]
+    files = lines[3:]
+    return title, author, author_email, files, url
+
+
+def is_system_probe(owners, files):
+    target = {
+        ("TEAM", "@DataDog/Networks"),
+        ("TEAM", "@DataDog/universal-service-monitoring"),
+        ("TEAM", "@DataDog/ebpf-platform"),
+        ("TEAM", "@DataDog/agent-security"),
+    }
+    for f in files:
+        match_teams = set(owners.of(f)) & target
+        if len(match_teams) != 0:
+            return True
+
+    return False
+
+
+@task
+def changelog(ctx, new_commit_sha):
+    old_commit_sha = ctx.run(
+        "aws ssm get-parameter --region us-east-1 --name "
+        "ci.datadog-agent.gitlab_changelog_commit_sha --with-decryption --query "
+        "\"Parameter.Value\" --out text",
+        hide=True,
+    ).stdout.strip()
+    if not new_commit_sha:
+        print("New commit sha not found, exiting")
+        return
+    if not old_commit_sha:
+        print("Old commit sha not found, exiting")
+        return
+    print(f"Generating changelog for commit range {old_commit_sha} to {new_commit_sha}")
+    commits = ctx.run(f"git log {old_commit_sha}..{new_commit_sha} --pretty=format:%h", hide=True).stdout.split("\n")
+    owners = read_owners(".github/CODEOWNERS")
+    messages = []
+
+    for commit in commits:
+        # see https://git-scm.com/docs/pretty-formats for format string
+        commit_str = ctx.run(f"git show --name-only --pretty=format:%s%n%aN%n%aE {commit}", hide=True).stdout
+        title, author, author_email, files, url = parse(commit_str)
+        if is_system_probe(owners, files):
+            author_handle = ctx.run(f"email2slackid {author_email.strip()}", hide=True).stdout
+            if author_handle:
+                author_handle = f"<@{author_handle}>"
+            else:
+                author_handle = author_email
+            time.sleep(1)  # necessary to prevent slack/sdm API rate limits
+            message_link = f"• <{url}|{title}>"
+            if "dependabot" not in author_email and "github-actions" not in author_email:
```

Comment:
This `if` could go much higher after the `parse` to prevent unnecessary work:
```
if "dependabot" in author_email or "github-actions" in author_email:
    continue
```

Reply:
I think we still want to include the commit here, I was doing this to exclude the author info because it's a bot account that has a really long email

Reply:
I cleaned the logic up slightly though

Reply:
You probably don't want to run `email2slackid` on it, rightt?

Reply:
ah I see what you mean.. updated